### PR TITLE
(POD-8) Add timeout_notification config to log warning before vm is destroyed.

### PR DESCRIPTION
**Pull request by isaac-hammes**

### CODE REVIEW

**Review by e-gris — APPROVED** [review]
Refactoring suggestion wholly optional.

File: lib/vmpooler/pool_manager.rb [hunk 1]
```diff
@@ -122,7 +122,7 @@ def remove_nonexistent_vm(vm, pool, redis)
       $logger.log('d', "[!] [#{pool}] '#{vm}' no longer exists. Removing from pending.")
     end
 
-    def fail_pending_vm(vm, pool, timeout, redis, exists: true)
+    def fail_pending_vm(vm, pool, timeout, timeout_notification, redis, exists: true)
```
[the remark this answers is not in the document]
Comment (e-gris) [reply lib/vmpooler/pool_manager.rb:125]: After further thought, I would suggest handling the !exists case first.

[the remark this answers is not in the document]
Comment (isaac-hammes) [reply lib/vmpooler/pool_manager.rb:125]: @e-gris Do you think this looks good? I think it looks quite a bit better now and is simplified.

[the remark this answers is not in the document]
Comment (e-gris) [reply lib/vmpooler/pool_manager.rb:125]: :+1:

**Pull request merged**